gnocchi/storage/ceph.py

```diff
@@ -91,19 +91,20 @@ def _store_metric_splits(self, metric, keys_and_data_and_offset,
             self.ioctx.operate_write_op(
                 op, self._build_unaggregated_timeserie_path(metric, 3))
 
-    def _delete_metric_measures(self, metric, key, aggregation, version=3):
-        name = self._get_object_name(metric, key, aggregation, version)
+    def _delete_metric_splits(self, metric, keys, aggregation, version=3):
+        with rados.WriteOpCtx() as op:
+            for key in keys:
+                name = self._get_object_name(metric, key, aggregation, version)
 
-        try:
-            self.ioctx.remove_object(name)
-        except rados.ObjectNotFound:
-            # It's possible that we already remove that object and then crashed
-            # before removing it from the OMAP key list; then no big deal
-            # anyway.
-            pass
+                try:
+                    self.ioctx.remove_object(name)
+                except rados.ObjectNotFound:
+                    # It's possible that we already remove that object and then
+                    # crashed before removing it from the OMAP key list; then
+                    # no big deal anyway.
+                    pass
+                self.ioctx.remove_omap_keys(op, (name,))
```

Comment:
why not just pass in a tuple of all names? or does rados figure this out?

Reply:
You're right, I missed that!